sdk/nodejs/cmd/pulumi-language-nodejs/main.go

```diff
@@ -1484,7 +1520,42 @@ func (host *nodeLanguageHost) RunPlugin(
 	cmd.Dir = req.Pwd
 	cmd.Env = env
 	cmd.Stdout, cmd.Stderr = stdout, stderr
-	if err := cmd.Run(); err != nil {
+
+	run := func() error {
+		if err := cmd.Start(); err != nil {
+			return err
+		}
+		if req.GetAttachDebugger() {
+			cfg := map[string]any{
```

Comment:
This block is really similar to line 828 and the program block, can they just share with a different "name"

Reply:
Done, factored this and the above out into separate functions.  Thanks!